### DISCUSSION TITLE
Async arrow function loses `this` context

### DISCUSSION THREAD

**Discussion by tinder-tannerbennett** [microsoft/vscode-discussions#953] [Extension Development QnA]
- macOS Ventura 13.6.2
- VS Code 1.84.2 (Electron 25.9.2)
- `yo` version: 5.0.0
- Issue appears with and without webpack

On my machine, in a brand new extension, using an async arrow function causes the arrow func to lose the `this` context:

```ts
class Foo {
    bar() {
        const nothis = async () => {
            console.log(this); // undefined
        };
        nothis();
        
        const hasthis = () => {
            console.log(this); // Foo
        };
        hasthis();
    }
}

...

const f = new Foo();
f.bar();
```

In the browser, they both have `this`. In an extension created on other machines, `this` is defined. But on my machine in a VS Code extension I wrote, it is always `undefined`.

What could this _possibly_ be?

### Update

This appears to be a bug in VS Code or whatever debugger it uses. Turns out the debugger is only _showing_ me that `this = undefined`. In reality it is there! Silly me, thinking I could trust the variables view over `console.log` or the actual debug console.

Steps to reproduce:

1. Download VS Code / update to latest version
2. Create a new TS extension with the command `yo code` (install `yo` with `npm install -g yo generator-code`)
3. Open it in VS Code and paste the above code in `extension.ts` with the class at the top and the code below the `...` in `activate()`
4. Place a breakpoint inside each of the arrow funcs
5. Run the watch build task (⌘⇧P → _Run Task_ → _npm watch_)
6. Run and debug the extension (⌘⇧D → spacebar)
7. Run the _Hello World_ command (⌘⇧P → _Hello World_)
8. Observe the debugger at each breakpoint. Hover over `this` and see that it is truly there

**Comment by olguzzar** [comment]
Hi @tinder-tannerbennett, thanks for the detailed repro steps! Since this seems to be a general bug with VS Code, please file an issue in the [main VS Code repo](https://github.com/microsoft/vscode/issues). Going to close this discussion out for now!
